internal/nearcache/repairing_task.go

```diff
@@ -0,0 +1,568 @@
+	"github.com/hazelcast/hazelcast-go-client/internal/cluster"
+	"github.com/hazelcast/hazelcast-go-client/internal/invocation"
+	ilogger "github.com/hazelcast/hazelcast-go-client/internal/logger"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/codec"
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+/*
+ReparingTask runs on Near Cache side and only one instance is created per data-structure type like IMap and ICache.
+Repairing responsibilities of this task are:
+
+    * To scan RepairingHandlers to see if any Near Cache needs to be invalidated according to missed invalidation counts (controlled via MaxToleratedMissCount).
+    * To send periodic generic-operations to cluster members in order to fetch latest partition sequences and UUIDs (controlled via InvalidationMinReconciliationIntervalSeconds.
+
+See: com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask
+*/
+type ReparingTask struct {
+	reconciliationIntervalNanos int64
+	lastAntiEntropyRunNanos     int64
+	handlers                    *sync.Map
+	ss                          *serialization.Service
+	ps                          *cluster.PartitionService
+	doneCh                      <-chan struct{}
+	invalidationMetaDataFetcher InvalidationMetaDataFetcher
+	lg                          ilogger.LogAdaptor
+	maxToleratedMissCount       int
+	partitionCount              int32
+	running                     int32
+}
+
+func NewReparingTask(recInt int, maxMissCnt int, ss *serialization.Service, ps *cluster.PartitionService, lg ilogger.LogAdaptor, mf InvalidationMetaDataFetcher, doneCh <-chan struct{}) *ReparingTask {
+	nc := &ReparingTask{
+		reconciliationIntervalNanos: int64(recInt) * 1_000_000_000,
+		maxToleratedMissCount:       maxMissCnt,
+		invalidationMetaDataFetcher: mf,
+		doneCh:                      doneCh,
+		ss:                          ss,
+		ps:                          ps,
+		lg:                          lg,
+		handlers:                    &sync.Map{},
+		partitionCount:              ps.PartitionCount(),
+	}
+	return nc
+}
+
+func (rt *ReparingTask) start() {
+	// see: com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask#scheduleNextRun
+	rt.lg.Debug(func() string {
+		return "ReparingTask started"
+	})
+	timer := time.NewTicker(1 * time.Second)
+	defer timer.Stop()
+	rt.run()
+	for {
+		select {
+		case <-rt.doneCh:
+			return
+		case <-timer.C:
+			rt.run()
+		}
+	}
+}
+
+func (rt *ReparingTask) RegisterAndGetHandler(ctx context.Context, name string, nc *NearCache) (RepairingHandler, error) {
+	handler := NewRepairingHandler(name, nc, rt.partitionCount, rt.ss, rt.ps, rt.lg)
+	// ignoring the "loaded" return value
+	h, loaded := rt.handlers.LoadOrStore(name, handler)
+	if !loaded {
+		if err := rt.invalidationMetaDataFetcher.Init(ctx, handler); err != nil {
+			return RepairingHandler{}, nil
+		}
```

Comment:
This is done after setting staleReadDetector in Java. Just to be sure that there is no behaviour change can you do the same? Also, won't you implement the fallback initializer `initRepairingHandlerAsync`